bundles/org.openhab.binding.zm/src/main/java/org/openhab/binding/zm/internal/handler/ZmBridgeHandler.java

```diff
@@ -0,0 +1,645 @@
+        if (!zmAuth.isAuthorized()) {
+            return null;
+        }
+        try {
+            List<String> parameters = new ArrayList<>();
+            parameters.add("sort=StartTime");
+            parameters.add("direction=desc");
+            parameters.add("limit=1");
+            String response = executeGet(
+                    buildUrlWithParameters(String.format("/api/events/index/MonitorId:%s.json", id), parameters));
+            EventsDTO events = GSON.fromJson(response, EventsDTO.class);
+            if (events != null && events.eventsList != null && events.eventsList.size() == 1) {
+                EventDTO e = events.eventsList.get(0).event;
+                Event event = new Event(e.eventId, e.name, e.cause, e.notes, e.startTime, e.endTime);
+                event.setFrames(e.frames);
+                event.setAlarmFrames(e.alarmFrames);
+                event.setLength(e.length);
+                return event;
+            }
+        } catch (JsonSyntaxException e) {
+            logger.debug("Bridge: JsonSyntaxException: {}", e.getMessage(), e);
+        }
+        return null;
+    }
+
+    private @Nullable VersionDTO getVersion() {
+        if (!zmAuth.isAuthorized()) {
+            return null;
+        }
+        VersionDTO version = null;
+        try {
+            String response = executeGet(buildUrl("/api/host/getVersion.json"));
+            version = GSON.fromJson(response, VersionDTO.class);
+        } catch (JsonSyntaxException e) {
+            logger.debug("Bridge: JsonSyntaxException: {}", e.getMessage(), e);
+        }
+        return version;
+    }
+
+    private void setAlarm(String url) {
+        executeGet(url);
+    }
+
+    @SuppressWarnings("null")
+    private MonitorState getState(String id) {
+        if (!zmAuth.isAuthorized()) {
+            return MonitorState.UNKNOWN;
+        }
+        try {
+            String response = executeGet(buildUrl(String.format("/api/monitors/alarm/id:%s/command:status.json", id)));
+            MonitorStateDTO monitorState = GSON.fromJson(response, MonitorStateDTO.class);
+            if (monitorState != null) {
+                MonitorState state = monitorState.state;
+                return state != null ? state : MonitorState.UNKNOWN;
+            }
+        } catch (JsonSyntaxException e) {
+            logger.debug("Bridge: JsonSyntaxException: {}", e.getMessage(), e);
+        }
+        return MonitorState.UNKNOWN;
+    }
+
+    public @Nullable String executeGet(String url) {
+        try {
+            long startTime = System.currentTimeMillis();
+            String response = HttpUtil.executeUrl("GET", url, API_TIMEOUT_MSEC);
+            logger.trace("Bridge: Http GET of '{}' returned '{}' in {} ms", url, response,
+                    System.currentTimeMillis() - startTime);
+            return response;
+        } catch (IOException e) {
+            logger.debug("Bridge: IOException on url '{}': {}", url, e.getMessage(), e);
```

Comment:
The stack trace should only be logged if something unexpected happened, like the detection of a bug in your code. Same for below. A plain log message should be enough when the network fails.
```suggestion
            logger.debug("Bridge: IOException on url '{}': {}", url, e.getMessage());
```

Reply:
Fixed